### PR TITLE
net: lib: nrf_cloud: rename nct_ to nrfcloud_transport_

### PR DESCRIPTION
**Pull request by DematteisGiacomo**
Reason is conflicting naming with zephyr native simulator nct_if.h

### CODE REVIEW

**Comment by jayteemo** [comment]
what is the actual conflict you are fixing?
is it only `nct_init`?
if so, can you just change the cloud function to `int nct_initialize(const char * const client_id)`

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_log_backend.c [hunk 1]
```diff
@@ -257,7 +257,7 @@ static void logger_process(const struct log_backend *const backend, union log_ms
 
 	logs_init_context(rest_ctx, device_id, level, src_id, src_name, dom_id, ts, &log_context);
 
-	log_output_func = log_format_func_t_get(log_format_current);
+	log_output_func = log_format_funrfct_get(log_format_current);
```
Comment (jorgenmk) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_log_backend.c:260]: This doesn't look correct.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c [hunk 2]
```diff
@@ -236,19 +236,19 @@ static bool strings_compare(const char *s1, const char *s2, uint32_t s1_len,
 }
 
 /* Verify if the RX topic is a control channel topic or not. */
-static bool nrf_cloud_cc_rx_topic_decode(const struct mqtt_topic *topic, enum nct_cc_opcode *opcode)
+static bool nrf_cloud_cc_rx_topic_dec(const struct mqtt_topic *topic, enum nrfct_cc_opcode *opcode)
```
Comment (jorgenmk) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c:239]: Why this change? Is it related to max line length?

**Review by jorgenmk — CHANGES_REQUESTED** [review]

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_log_backend.c (hunk 1 above)
Comment (DematteisGiacomo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_log_backend.c]: it is not fixed :+1:

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c (hunk 2 above)
Comment (DematteisGiacomo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c:239]: yes, it is due to max line

**Comment by DematteisGiacomo** [comment]
> what is the actual conflict you are fixing? is it only `nct_init`? if so, can you just change the cloud function to `int nct_initialize(const char * const client_id)`

@jayteemo  yes it is only `nct_init`, I'll just rename that one then. 
The idea for a general rename was to avoid future possible conflicts.

**Review by jayteemo — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Review by jorgenmk — APPROVED** [review]
